twitchio/websocket.py

```diff
@@ -181,7 +199,7 @@ def _task_callback(self, data, task):
             log.error("Authentication error. Please check your credentials and try again.")
             self._close()
         elif exc:
-            asyncio.create_task(self.event_error(exc, data))
+            self._background_tasks.append(asyncio.create_task(self.event_error(exc, data)))
```

Comment:
Due to the nature of your cleaning up cancelling all the tasks, you'll need to wrap the event_error in asyncio.shield

Reply:
done